app/src/main/java/com/stevesoltys/seedvault/transport/backup/PackageService.kt

```diff
@@ -159,7 +164,13 @@ internal fun PackageInfo.isSystemApp(): Boolean {
 
 internal fun PackageInfo.allowsBackup(): Boolean {
     if (packageName == MAGIC_PACKAGE_MANAGER || applicationInfo == null) return false
-    return applicationInfo.flags and FLAG_ALLOW_BACKUP != 0
+
+    // At backup time, the system will filter out any apps that *it* does not want to be backed up.
+    // Now that we have switched to D2D, *we* generally want to back up as much as possible;
+    // part of the point of D2D is to ignore FLAG_ALLOW_BACKUP (allowsBackup). So, we return true.
+    // See frameworks/base/services/backup/java/com/android/server/backup/utils/
+    // BackupEligibilityRules.java lines 74-81 and 163-167 (tag: android-13.0.0_r8).
+    return true
```

Comment:
If we always return `true` here, I wonder, how that influences the other methods in this class that make use of it.

For example, will we still ever show "Apps that do not allow data backup" in the UI? If not, can we remove the related UI code and strings already?

Do we need to update the comment in other methods that still refer to backup not being allowed?

If we *almost* always return `true` here, could we maybe just eliminate this method completely? Maybe even other methods that are calling it, or at least simplifying them?

Sorry for being a pain in the ass here, but I think it is important that when implementing a new feature, it goes all the way. Just hacking the important bits in quickly will amass technical debt for future developers to solve.

Reply:
Hi, since this is the 'do it properly' branch, i'd like to give my input here: i'd like to include all apps, but maybe, not everyone does. Especially since i know that app developers (eg of davx5) set allowBackup to false, as they did now want to give the false impression their app would work perfectly after a backup. I'd favor (in a polished, perfect solution) an option to say something like 'force include all apps'.